dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs

```diff
@@ -57,16 +57,15 @@ public async Task<Plan> CreatePlanAsync(string goal, CancellationToken cancellat
             throw new SKException("The goal specified is empty");
         }
 
-        string relevantFunctionsManual = await this._context.GetFunctionsManualAsync(goal, this.Config, cancellationToken).ConfigureAwait(false);
-        this._context.Variables.Set("available_functions", relevantFunctionsManual);
+        string relevantFunctionsManual = await this._kernel.CreateNewContext().GetFunctionsManualAsync(goal, this.Config, cancellationToken).ConfigureAwait(false);
 
-        this._context.Variables.Update(goal);
+        ContextVariables vars = new(goal) { ["available_functions"] = relevantFunctionsManual };
 
-        var planResult = await this._functionFlowFunction.InvokeAsync(this._context, cancellationToken: cancellationToken).ConfigureAwait(false);
+        var planResult = await this._kernel.RunAsync(this._functionFlowFunction, vars, cancellationToken).ConfigureAwait(false);
```

Comment:
Ahh, this is breaking some of the UTs that only mocked the InvokeAsync. You'll need to mock out `RunAsync` for this.